paymentsheet/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -5,284 +5,267 @@ import android.view.LayoutInflater
+                        showCheckboxControlledFields = sheetViewModel.newPaymentSelection?.let {
+                            sheetViewModel.newPaymentSelection?.customerRequestedSave ==
+                                PaymentSelection.CustomerRequestedSave.RequestReuse
+                        } ?: layoutFormDescriptor.showCheckboxControlledFields,
+                        merchantName = sheetViewModel.merchantName,
+                        amount = sheetViewModel.amount.value,
+                        billingDetails = sheetViewModel.config?.defaultBillingDetails,
+                        shippingDetails = sheetViewModel.config?.shippingDetails,
+                        injectorKey = sheetViewModel.injectorKey,
+                        initialPaymentMethodCreateParams =
+                        sheetViewModel.newPaymentSelection?.paymentMethodCreateParams?.typeCode
+                            ?.takeIf {
+                                it == selectedItem.code
+                            }?.let {
+                                when (val selection = sheetViewModel.newPaymentSelection) {
+                                    is PaymentSelection.New.GenericPaymentMethod ->
+                                        selection.paymentMethodCreateParams
+                                    is PaymentSelection.New.Card ->
+                                        selection.paymentMethodCreateParams
+                                    else -> null
                                 }
-                            )
-                        }
+                            }
                     )
                 }
-            }
-        }
-
-        sheetViewModel.isResourceRepositoryReady.observe(viewLifecycleOwner) { isReady ->
-            if (isReady == true) {
-                val selectedPaymentMethodIndex = paymentMethods.indexOf(
-                    sheetViewModel.addFragmentSelectedLPM
-                ).takeUnless { it == -1 } ?: 0
 
-                // In order to be able to read sheetViewModel.addFragmentSelectedLPM
-                // the repository needs to be set. This might not be set if recovery
-                // from a killed process or don't keep activities event
-                if (paymentMethods.size > 1) {
-                    setupRecyclerView(
-                        viewBinding,
-                        paymentMethods,
-                        sheetViewModel.addFragmentSelectedLPM
+                LaunchedEffect(arguments) {
+                    elementsFlow.emit(
+                        getElements(
+                            requireContext(),
+                            arguments,
+                            sheetViewModel.lpmResourceRepository,
+                            sheetViewModel.addressResourceRepository
+                        )
+                    )
+                    showCheckboxFlow.emit(
+                        arguments.showCheckbox
                     )
                 }
```

Comment:
Can we refactor this so that `PaymentElement` takes in simple values instead of flows?